apps/server/src/modules/authorization/README.md

```diff
@@ -0,0 +1,209 @@
+Each user has a role with permissions in the scope of the domain object they want to interact.
+Each authorisation required a user.
+
+#### system users
+
+We have console operations, or operations based on API_KEYS that are used between internal services for already authorised operations like copy and copy files in file service.
+For this we want to use system user and roles with own permissions.
+> They are not introduced for now
+
+## Rules
+
+Rules are the location that solve the authorisation for a domain object.
+A rule has implement a check for what they should match.
+They are called from the authorisation service.
+They must be support our actions.
+
+> We highly recoment that every single operation and check in the rule is implement as additional method to keep the logic clean and moveable
+
+## User Permissions vs Scope Based Permissions
+
+It exit for now as free permissions bind to a user without defined scope.
+It exist also scope based permissions to solved the situation, that a user can have in different (scope)roles in different (domain)scopes. For example in teams where the student can have teamuser in one team, or teamadminstrator in a other.
+
+> This is not what we want in future.
+> For now, we must handle scope permissions, system user and user as seperated special cases and implementations.
+> By implementation of user permissions as school scope and others as instance scope we can do it in one way for all situations
+
+## How should you Authorised a Operation?
+
+The only point is in use cases (UC). They must solve the authorisation and orchestrate the logic that should be done in services, or private methods.
+You should never implement authorisations on service level, to avoid different authorisation steps.
+By calling other micro service for already authorised operations please use API_KEYS.
+
+> Not implemented but coming soon.
+
+## How to use Authorisation Service
+
+### Example 1 - ...
+
+{{ add more }}
+
+### Example 2 - ...
+
+{{ add more }}
+
+### Example 3 - ...
+
+{{ add more }}
+
+## Structure of the Authorisation Components
+
+### feathers-* (legacy/deprecated)
+
+It exist a adapter to call featherJS endpoints that solve authorisations.
+> This service is only used in news and should not use on any other place.
+> We want to remove it completly.
+
+### Authorisation Module
+
+The authorsation module is the core of authorisation. They collect all needed informations and handle it for you behind a small interface. It export the authoriation service that can be used in your use case over injections.
+
+### Reference.loader
+
+For situations where only the id and the domain object (string) type is knowen, it is possible to use the *ByReferences methodes.
+They load the reference directly.
+> Please keep in mind that it can be have impact of the performance if you use it wrongly.
+> We do it as seperate method to avoid using in areas where the domain object should be exist, but it is easyer to only pass the id.
```

Comment:
```suggestion
> We keep it as a seperate method to avoid the usage in areas where the domain object should exist, because we see the risk that a developer could be tempted by the ease of only passing the id.
```